packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -415,23 +443,33 @@ class HttpRouteSpec extends RouteSpec {
   }
 
   public bind(scope: Construct): RouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
-    }
+    const pathMatchConfig = this.match?.path?.bind(scope);
 
+    // Set prefix path match to '/' if none of path matches are defined.
+    const prefixPathMatch = pathMatchConfig ? pathMatchConfig.prefixPathMatch : '/';
+    const wholePathMatch = pathMatchConfig?.wholePathMatch;
     const headers = this.match?.headers;
-    validateMatchArrayLength(headers);
+    const queryParameters = this.match?.queryParameters;
+
+    if (prefixPathMatch && prefixPathMatch[0] !== '/') {
+      throw new Error(`Prefix Path for the match must start with \'/\', got: ${prefixPathMatch}`);
+    }
+    if (wholePathMatch?.exact && wholePathMatch.exact[0] !== '/') {
+      throw new Error(`Exact Path for the match must start with \'/\', got: ${wholePathMatch.exact}`);
+    }
+    validateHttpMatchArrayLength(headers, queryParameters);
 
     const httpConfig: CfnRoute.HttpRouteProperty = {
       action: {
         weightedTargets: renderWeightedTargets(this.weightedTargets),
       },
       match: {
-        prefix: prefixPath,
+        prefix: prefixPathMatch,
+        path: wholePathMatch,
         headers: headers?.map(header => header.bind(scope).headerMatch),
         method: this.match?.method,
         scheme: this.match?.protocol,
+        queryParameters: queryParameters?.map(queryParameter => queryParameter.bind(scope).queryParameterMatch),
       },
```

Comment:
Here's an idea: let's leave `match` as `undefined` if either `this.match` is `undefined`, or if all of its properties are `undefined`.

So, this would be:

```suggestion
      match: allPropertiesAreUndefined(this.match) ? undefined : {
        prefix: pathMatchConfig.prefixPathMatch,
        path: pathMatchConfig.wholePathMatch,
        headers: headers?.map(header => header.bind(scope).headerMatch),
        method: this.match?.method,
        scheme: this.match?.protocol,
        queryParameters: queryParameters?.map(queryParameter => queryParameter.bind(scope).queryParameterMatch),
      },
```

Reply:
Actually [`match`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-route-httproute.html) is the required property so would not be able to set `undefined`.

In addition, we set `prefix` to `/` if neither `prefix` and `path` are defined so we are guaranteed to have at least one property be defined.

Reply:
I see all of the properties of the `CfnGatewayRoute.HttpGatewayRouteMatchProperty` are opitonal.

Is there some CloudFormation validation that errors out if an empty object is passed as its value, or is it legal to just omit all of the properties? Because if it's legal, let's just do that.

Reply:
> Is there some CloudFormation validation that errors out if an empty object is passed as its value

Yes, CloudFormation has a validation to prohibit passing empty value or empty array.

> is it legal to just omit all of the properties

No, CloudFormation also validates if least one type of match is specified.

Reply:
Since CDK has a validation to check empty array and non-array property accepts an object so I believe CDK-side has all validation that are needed.